marlowe-dashboard-client/src/Capability/Contract.purs

```diff
@@ -0,0 +1,67 @@
+module Capability.Contract
+  ( class MonadContract
+  , activateContract
+  , getContractInstance
+  , invokeEndpoint
+  , getWalletContractInstances
+  , getAllContractInstances
+  , getContractDefinitions
+  ) where
+
+import Prelude
+import AppM (AppM)
+import Bridge (toFront)
+import Capability.Ajax (runAjax)
+import Control.Monad.Except (lift)
+import Data.Newtype (unwrap)
+import Data.RawJson (RawJson)
+import Data.UUID (toString) as UUID
+import Halogen (HalogenM)
+import MainFrame.Types (WebData)
+import Plutus.PAB.Effects.Contract.ContractExe (ContractExe)
+import Plutus.PAB.Webserver (getApiNewContractDefinitions, getApiNewContractInstanceByContractinstanceidStatus, getApiNewContractInstances, getApiNewContractInstancesWalletByWalletid, postApiNewContractActivate, postApiNewContractInstanceByContractinstanceidEndpointByEndpointname)
+import Plutus.PAB.Webserver.Types (ContractActivationArgs, ContractInstanceClientState, ContractSignatureResponse)
+import Types (ContractInstanceId)
+import WalletData.Types (Wallet)
+
+-- The PAB PSGenerator (using Servant.PureScript) automatically generates a PureScript module with
+-- functions for calling all PAB API endpoints. This `MonadContract` class wraps these up in a
+-- 'capability' monad (https://thomashoneyman.com/guides/real-world-halogen/push-effects-to-the-edges/)
+-- with some nicer names, and mapping the result to RemoteData.
+class
+  Monad m <= MonadContract m where
+  activateContract :: ContractActivationArgs ContractExe -> m (WebData ContractInstanceId)
+  getContractInstance :: ContractInstanceId -> m (WebData ContractInstanceClientState)
+  invokeEndpoint :: RawJson -> ContractInstanceId -> String -> m (WebData Unit)
```

Comment:
I think `invokeEndpoint` is low level and probably should not be exposed by this API

Reply:
I'm intending this module to be low level, a pretty thin wrapper around the automatically generated API module that's nicer to use. Ultimately it would live in a web-common directory and be used by the pab-client as well. The `Capability.Marlowe` module is intended as the next layer up, and the one we'd mainly use here.